doc/modules/decomposition.rst

```diff
@@ -278,6 +257,88 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. _Randomized_kPCA:
+
+Kernel PCA using randomized SVD
+-------------------------------
+
+In :ref:`kernel_PCA`, the number of components found is equal to the number of
+samples. Many real-world datasets have large number of samples ! In these cases
+finding *all* the components with a full kPCA is a waste of computation time,
+as data is mostly described by the first few components
+(``n_components<=100``).
+
+The optional parameter ``eigen_solver='randomized'`` can be used to
+*drastically* reduce the computation time when the number of requested
+``n_components`` is small compared with the number of samples.
+This is done automatically by default if you do not specify any
+``eigen_solver`` and if ``n_components`` is less than 80% of
+the number of samples, see :class:`KernelPCA` for details.
+
+For example the figure below shows the same "circles" dataset (top left)
+reconstructed with various approximations (number of components decreases from
+2000 - full - to 4). You can compare the execution times vs. the reconstruction
+quality :
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_approximate_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca_approximate.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca_approximate.py`
+
+Note: this technique is the same than in :ref:`RandomizedPCA`.
+
+Kernel PCA solvers benchmarks
+-----------------------------
```

Comment:
We usually don't include benchmarks in the UG. Guidelines as you did above are good enough

Reply:
It is a bit sad, as these benchmarks would then exist but noone would know about the code being there. How can we make this at least visible so that people can quickly find the code if needed ?

Reply:
benchmarks will get more visibility once we have a dedicated benchmark suite, but for now these should be kept out of the UG.

Reply:
ok, I'll remove them.